### PR TITLE
Fix load-balanced TPU Splash masks

### PR DESCRIPTION
**Pull request by huytransformer**
# Description

Fixes TPU Splash local sliding masks when CP load balancing is enabled. The causal mask already uses the load-balanced query order, but the local sliding mask used the default query order. This PR adds a load-balanced local mask so both masks use the same query positions.

BUGS: b/416110599

# Tests

Ran:

```bash
python3 -m pytest tests/unit/attention_test.py -k LoadBalancedMaskTest
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by Shuwen-Fang — APPROVED** [review]

File: src/maxtext/layers/attention_op.py [hunk 1]
```diff
@@ -2163,6 +2171,13 @@ def __call__(
       return prefill_unnormalized_output / prefill_exponentials_sum
 
 
+def _load_balanced_q_sequence(shape: tuple[int, int], cp_size: int):
+  """Reorders query positions the same way as load-balanced input tokens."""
+  arr = np.arange(shape[0])
+  out = max_utils.reorder_mask_load_balancing(arr[None, :, None, None], cp_size, 1)
+  return out[0, :, 0, 0]
```
Comment (NuojCheng) [inline src/maxtext/layers/attention_op.py]: i feel sad reading line 2177 and 2178. reorder_mask_load_balancing only works for 4D tensor?

**Review by NuojCheng — APPROVED** [review]

File: src/maxtext/layers/attention_op.py [hunk 2]
```diff
@@ -2223,3 +2232,23 @@ def __hash__(self):
...
+
+class LoadBalancedLocalMask(splash_attention_mask.LocalMask):
+  """Lazy local mask using load-balanced query order."""
+
+  def __init__(
+      self,
+      shape: tuple[int, int],
+      window_size: tuple[int | None, int | None],
+      offset: int,
+      shard_count: int = 1,
```
Comment (gobbleturk) [inline src/maxtext/layers/attention_op.py:2244]: what is the difference betwen shard_count and cp_size? Can you include comments for each property?

File: src/maxtext/layers/attention_op.py [hunk 3]
```diff
@@ -2223,3 +2232,23 @@ def __hash__(self):
...
+      cp_size: int = 4,
```
Comment (gobbleturk) [inline src/maxtext/layers/attention_op.py]: I would prefer defaulting cp_size to -1 so users are forced to set it

**Review by gobbleturk — APPROVED** [review]

File: src/maxtext/layers/attention_op.py (hunk 1 above)
Comment (huytransformer) [reply src/maxtext/layers/attention_op.py]: noo don't be sad it was copy&paste from 2196-2200. fixed :)

File: src/maxtext/layers/attention_op.py (hunk 3 above)
Comment (huytransformer) [reply src/maxtext/layers/attention_op.py]: mm actually that's a good idea. done!!
